BitmovinAnalyticsCollector/Classes/Collector/analytics/BitmovinAnalyticsInternal.swift

```diff
@@ -71,8 +74,14 @@ extension BitmovinAnalyticsInternal: StateMachineDelegate {
         sendEventData(eventData: eventData)
     }
 
-    func stateMachineDidEnterError(_ stateMachine: StateMachine) {
+    func stateMachineDidEnterError(_ stateMachine: StateMachine, data: [AnyHashable: Any]?) {
         let eventData = createEventData(duration: 0)
+        if((data?[BitmovinAnalyticsInternal.ErrorCodeKey]) != nil) {
```

Comment:
Please unwrap the `data` with the `if let` structure to get rid of that optional madness :D